src/cloudabi-run/yaml_file_descriptor_factory.cc

```diff
@@ -55,6 +55,10 @@ const argdata_t *YAMLFileDescriptorFactory::GetScalar(const YAML::Mark &mark,
       throw YAML::ParserException(mark, "XXX!");
     }
     return argdatas_.emplace_back(argdata_t::create_fd(fd)).get();
+  } else if (tag == "tag:nuxi.nl,2015:cloudabi/executable") {
+    if (execfd == -1)
```

Comment:
This check won't be needed if the file descriptor is passed in through te constructor :-)